src/transformers/pipelines/__init__.py

```diff
@@ -905,13 +927,29 @@ def pipeline(
 
     model_config = model.config
     hub_kwargs["_commit_hash"] = model.config._commit_hash
+
     load_tokenizer = (
         type(model_config) in TOKENIZER_MAPPING
         or model_config.tokenizer_class is not None
         or isinstance(tokenizer, str)
     )
-    load_feature_extractor = type(model_config) in FEATURE_EXTRACTOR_MAPPING or feature_extractor is not None
-    load_image_processor = type(model_config) in IMAGE_PROCESSOR_MAPPING or image_processor is not None
+    load_feature_extractor = (
+        type(model_config) in FEATURE_EXTRACTOR_MAPPING
+        or feature_extractor is not None
+        or isinstance(feature_extractor, str)
+    )
+    load_image_processor = (
+        type(model_config) in IMAGE_PROCESSOR_MAPPING
+        or image_processor is not None
+        or isinstance(image_processor, str)
+    )
+    load_processor = type(model_config) in PROCESSOR_MAPPING or processor is not None or isinstance(processor, str)
```

Comment:
And here!

Reply:
addressed in https://github.com/huggingface/transformers/pull/32514/commits/d553dab8a33c4144b29c649115fb3a41cd92d6bf